docs/developer-env.md

```diff
@@ -0,0 +1,473 @@
+
+### Development Containers extension
+
+The VS Code Dev Containers extension lets you use a container as a full-featured development environment. It allows you to open any folder inside (or mounted into) a container and take advantage of VS Code's full feature set.
+
+The Dev Containers extension is included as a part of the [Remote Development extension pack](https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.vscode-remote-extensionpack). See {ref}`remote-development-ext` for installation instructions.
+
+### Create and run a dev container
+
+A `devcontainer.json` file in your project directory instructs VS Code how to access, create, and configure a development container. It can be used to run an application or to separate tools, libraries, or runtimes.
+
+The `devcontainer.json` can be used to:
+
+- Install additional tools in the container.
+- Automatically install extensions.
+- Forward or publish additional ports.
+- Set runtime arguments.
+- Reuse or extend your existing Docker Compose setup.
+- Add more Advanced container configuration.
+
+To create a dev container with an existing image:
+
+1. Create `.devcontainer/devcontainer.json` in the root of your project.
+1. Add an image with the Nextflow tooling and VS Code customizations to the `devcontainer.json`. For example:
+
+    ```json
+    {
+        "name": "Nextflow Dev Container",
+        "image": "nfcore/gitpod:latest",
+        "remoteUser": "vscode",
+        "runArgs": ["--privileged"],
+
+        // Configure tool-specific properties.
+        "customizations": {
+            // Configure properties specific to VS Code.
+            "vscode": {
+                // Set *default* container specific settings.json values on container create.
+                "settings": {
+                    "python.defaultInterpreterPath": "/opt/conda/bin/python"
+                },
+
+                // Add the IDs of extensions you want installed when the container is created.
+                "extensions": ["ms-python.python", "ms-python.vscode-pylance", "nf-core.nf-core-extensionpack", "nextflow.nextflow"]
+            }
+        }
+    }
+    ```
+
+    :::{note}
+    Instead of using a prebuilt image, a custom Dockerfile may also live in the `.devcontainer` folder. You can replace the image property in `devcontainer.json` with dockerfile and utilize the custom container. See [Create a Dev Container](https://code.visualstudio.com/docs/devcontainers/create-dev-container) for more information.
+    :::
+
+1. Enter **Dev Containers: Reopen in Container** in the VS Code Command Palette and reopen your project. You should now see the name of the container ("Nextflow Dev Container" if using the above example) in the bottom left corner of VS Code.
+
+:::{note}
+Dev Containers can also be used by GitHub Codespaces in VS Code or the browser. See [GitHub Codespaces](https://code.visualstudio.com/docs/remote/codespaces) for more information.
```

Comment:
not sure this makes much sense in the context of nextflow workflow in which each process can have its own container. Our best practice is Conda + Wave for container reproducibility and portability

Reply:
Fair point. I'm not committed to keeping this. It's very thin on details and can get complicated quickly. I'll comment it out.

Reply:
Yeah agree with this. I think that devcontainers for the pipeline development environment is not incompatible with using conda / wave etc for actually running the pipeline.